definitions/features/dynflow_sidekiq.rb

```diff
@@ -22,18 +18,11 @@ def config_files
     ].flatten
   end
 
-  private
-
-  def instance_priority(instance)
-    # Orchestrator should be started before the workers are
-    instance.end_with?('@orchestrator') ? 30 : 31
-  end
-
-  def service_names
-    configured_instances.map { |instance| "dynflow-sidekiq@#{instance}" }
-  end
-
-  def configured_instances
-    Dir['/etc/foreman/dynflow/*'].map { |config| File.basename(config, '.yml') }
+  def services
+    [
+      system_service('dynflow-sidekiq@orchestrator', 30, :parent_unit => 'dynflow-sidekiq@'),
+      system_service('dynflow-sidekiq@worker-hosts-queue', 31, :parent_unit => 'dynflow-sidekiq@'),
+      system_service('dynflow-sidekiq@worker', 31, :parent_unit => 'dynflow-sidekiq@')
+    ]
```

Comment:
With new attribute `parent_unit` on `service`, just one additional option needs to be passed here. 

Example -

```
def services
   service_names.map { |service| system_service service, instance_priority(service), :parent_unit => 'dynflow-sidekiq@' }
end
```

Do you really need to remove existing private methods of this feature? Am I missing anything here?

Reply:
We had three methods,
1. The configured_instances was there to get all instances of service. Now we are predefined those service name(3 of them) with parent_unit attribute.
2. service_names was there to add common name "dynflow-sidekiq@" to all three, here also we have predefined services now so this is also not required.
3. instance_priority was to decide priority for these three services, and we have predefined priority too. so this is not required too.

Reply:
As long as dynflow-sidekiq has only these three & fixed configured_instances but I am not sure about number of dynflow-sidekiq service instances.

@johnpmitsch, will it be okay to remove dynamic construction for dynflow-sidekiq services and handle three instances?  any suggestion?

Reply:
One of the key benefits of the switch to dynflow-sidekiq from our custom executor (dynflowd) is the ability to scale up by just spawning more instances of service. If you remove the dynamic construction for dynflow-sidekiq services, then foreman-maintain will not be able to deal with scaled up installations, possibly leading to errors.

Also if I recall correctly `dynflow-sidekiq@worker-hosts-queue` is only present if katello is installed, so you would still have to make this somewhat dynamic.

Reply:
Thank you @adamruzicka for sharing details. very helpful!